Connecting_MonteCarlo_to_ModernAI/N2_2D_Ising_Model.ipynb

```diff
@@ -0,0 +1,1233 @@
+        "\n",
+        "        # then the total energy for the atom is calculated\n",
+        "        E_a = -0.5*lattice[r,c]*sum_NN\n",
+        "        \n",
+        "        # --> Step 3: Change the spin and re-calculate the energy\n",
+        "        # Remember we said above that changing the spin just changes the sign\n",
+        "        # of the energy\n",
+        "        E_b = -1*E_a\n",
+        "\n",
+        "        # --> Step 4: If the energy decreased or the probability sample \n",
+        "        # meets the requirement\n",
+        "        if E_b < E_a or np.exp(-(E_b - E_a)*(beta)) > np.random.rand():\n",
+        "            # Update the actual spin value in the lattice\n",
+        "            lattice[r, c] *= -1\n",
+        "\n",
+        "        # --> Step 5: The stop condition for this code block is that all of the\n",
+        "        # interior atoms have been updated at least once"
+      ],
+      "metadata": {
+        "id": "5im24DR11l50"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Next, I'm going to visualize my system again to see if anything changed.  Since I set the thermodynamic temperature at $\\beta=0 → T = \\infty$, it should still be in a mixed state."
+      ],
+      "metadata": {
+        "id": "M5zpUnh2GBw5"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "plt.imshow(lattice)\n",
+        "plt.colorbar()\n",
+        "plt.show()"
+      ],
+      "metadata": {
+        "id": "gLLMzQ6HUa06"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Above, we implemented a Monte Carlo Markov Chain (MCMC) to update the atom's spin state.  The state transition was considered based on sampling from both the Boltzmann distribution and the uniform distribution.  The MCMC simulation allowed us to implement the Hamiltonian model for how we think the sytem should behave."
+      ],
+      "metadata": {
+        "id": "QbTqKZCcG7PT"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Example Temperature Experiment\n",
+        "So far, so good!  \n",
+        "\n",
+        "Let's make some improvements to the code, and simulate the model for a range of temperatures."
+      ],
+      "metadata": {
+        "id": "mSHQkSk61lm4"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Making the lattice a little larger\n",
+        "sqrt_N = 25 \n",
+        "\n",
+        "# Create a new lattice\n",
+        "init_lattice = np.random.uniform(size=(sqrt_N,sqrt_N))\n",
+        "\n",
+        "#mask lattice\n",
+        "init_lattice[init_lattice>0.5]=1\n",
+        "init_lattice[init_lattice !=1]=-1\n",
+        "\n",
+        "# A new step here to create non-interacting atoms around the edge\n",
```

Comment:
Maybe you can add some text explanation to this adjustment? You discuss that it is a challenge before, so it might be clarifying for people to see written out exactly what you're doing to account for it.

Reply:
Thank you for your comment.  The in-line code comment has been updated as follows:

```
# A new step here to create non-interacting atoms around the edge by padding
# the array edges with zeroes.  This way, we can iterate over all N atoms in the 
# system without causing an out-of-bounds error
```